pyfar/dsp/dsp_interpolation.py

```diff
@@ -474,6 +474,143 @@ def fractional_delay_sinc(signal, delay, order=30, side_lobe_suppression=60,
     return signal
 
 
+def resample(signal, sampling_rate, match_amplitude="time", frac_limit=None):
+    """Resample signal to new sampling rate.
+
+    The SciPy function ``scipy.signal.resample_poly`` is used for resmapling.
+    Therefore, the factor ``L = sampling_rate/signal.sampling_rate``
+    is approximated by a fraction of two integer numbers. Therefore, the signal
+    first upsampled by an integer number given by the numerator of the fraction
+    , and downsampled by an integer value given by the denominator of the
+    fraction.
+
+    .. note ::
+
+        `sampling_rate` should be divisible by 10, otherwise it can cause an
+        infinite loop in the ``resample_poly`` function.
+
+        The amplitudes of the signal can match the amplitude of the input
+        signal in the time and frequency domain. See the parameter
+        `match_amplitude` and the examples for more information.
+
+    Parameters
+    ----------
+    signal : Signal
+        Input data to be resampled
+    sampling_rate : number
+        The new sampling rate in Hz
+    match_amplitude : string
+        Define the domain to match the amplitude of the resampled data.
+
+        ``'time'``
+            Maintains the amplitude in the time domain. This is useful for
+            recordings such as speech or music and must be used if
+            ``signal.signal_type = 'power'``.
+        ``'freq'``
+            Maintains the amplitude in the frequency domain by multiplying the
+            resampled signal by ``1/L`` (see above). This is often desired
+            when resampling impulse responses.
+
+        The default is ``'time'``.
+    frac_limit : int
+        Limit the denominator for approximating the resampling factor `L`
+        (see above). This can be used in case the resampling gets stuck in an
+        infinite loop (see note above) at the potenital cost of not exactly
+        realizing the target sampling rate.
+
+        The default is ``None``, which uses ``frac_limit = 1e6``.
+
+    Returns
+    -------
+    signal : pyfar.Signal
+        The resampled the input data.
```

Comment:
Should we add information about the length of the new signal? Should be ``up/down * signal.n_samples`` followed by a round, floor, or ceil operation (I suggested to introduce up and down above to make it more complete)